lib/src/run.dart

```diff
@@ -144,3 +152,56 @@ stderr:
 
 $stderr""";
 }
+
+
+/// Arguments passed to [Process.run] .
+/// See [Process.run] for more details.
+class RunOptions {
+  final String workingDirectory;
+  final Map<String, String> environment;
+  final bool includeParentEnvironment;
+  final bool runInShell;
+  final Encoding stdoutEncoding;
+  final Encoding stderrEncoding;
+
+  RunOptions({this.workingDirectory, this.environment,
+      this.includeParentEnvironment: true, this.runInShell: false,
+      this.stdoutEncoding: SYSTEM_ENCODING,
+      this.stderrEncoding: SYSTEM_ENCODING});
+
+  /// Create a clone when it's necessary to modify the passed runOptions to
+  /// avoid modifying the argument.
+  RunOptions clone() {
```

Comment:
Since all the fields are final, I'm not sure how you would update the returned clone with the changed values.  The pattern I've seen for this that I like is a method which returns an object with the same fields as an existing instance, except for explicit overrides.  It could either be a constructor:

(using ?? for brevity)

```
  RunOptions.from(RunOptions other, {String workingDirectory, Map<String, String> environment,
      bool includeParentEnvironment, bool runInShell,
      Encoding stdoutEncoding,
      Encoding stderrEncoding}) : this(
      workingDirectory: workingDirectory ?? other.workingDirectory,
      environment: environment ?? other.environment,
      includeParentEnvironment: includeParentEnvironment ?? other.includeParentEnvironment,
      stdoutEncoding: stdoutEncoding ?? other.stdoutEncoding,
      stderrEncoding: stderrEncoding ?? other.stderrEncoding
);
```

or an instance method:

```
  RunOptions with({String workingDirectory, Map<String, String> environment,
      bool includeParentEnvironment, bool runInShell,
      Encoding stdoutEncoding,
      Encoding stderrEncoding}) => new RunOptions(
      workingDirectory: workingDirectory ?? this.workingDirectory,
      environment: environment ?? this.environment,
      includeParentEnvironment: includeParentEnvironment ?? this.includeParentEnvironment,
      stdoutEncoding: stdoutEncoding ?? this.stdoutEncoding,
      stderrEncoding: stderrEncoding ?? this.stderrEncoding
);
```